cherrypy/test/test_patch_request.py

```diff
@@ -0,0 +1,75 @@
+"""Simple test for HTTP PATCH requests."""
+
+import cherrypy
+from cherrypy.test import helper
+
+class PatchRequestTests(helper.CPWebCase):
+
+    @staticmethod
+    def setup_server():
+        """Sets up a basic cherryPy server for testing.
+        """
+
+        class Root(object):
+            """Class to receive HTTP PATCH request. Has only 1 method 'PATCH'
+            which receives a request with a URI query parameter and a body
+            with data
+            """
+            exposed = True
+
+            def _cp_dispatch(self, vpath):
+                """Converts a URI from containing route parameters to a route 
+                containing a query string.
+
+                Example:
+                from: http://example.com/noun/value 
+                to  : http://example.com/noun?key=value
+
+                Args:
+                    vpath (str): The string of the
+
+                Returns:
+                    Root
+                """
+                
+                # since our routes will only contain the GUID, we'll only have 1 
+                # path. If we have more, just ignore it
+                if len(vpath) == 1:
+                    cherrypy.request.params['key'] = vpath.pop()
+                    
+                return self
+
+
+            def PATCH(self, **kwargs):
+                """Makes sure that PATCH requests work
+
+                Args:
+                    kwargs (dict): values passed in the URI
+
+                Returns:
+                    str: the values from the URI and body
+                """
+
+                data = cherrypy.request.body.read()
```

Comment:
@webknjaz 
Have I misunderstood the test? 

I save the body of the PATCH request, add it the route parameter I pulled from the URI, and compare it some known output. Isn't that right? Or, do I have it wrong?